examples/with_winit/src/lib.rs

```diff
@@ -0,0 +1,433 @@
+    #[cfg(target_arch = "wasm32")] render_state: RenderState,
+) {
+    use winit::{event::*, event_loop::ControlFlow};
+    let mut renderers: Vec<Option<Renderer>> = vec![];
+    #[cfg(not(target_arch = "wasm32"))]
+    let mut render_cx = render_cx;
+    #[cfg(not(target_arch = "wasm32"))]
+    let mut render_state = None::<RenderState>;
+    #[cfg(target_arch = "wasm32")]
+    let mut render_state = {
+        renderers.resize_with(render_cx.devices.len(), || None);
+        let id = render_state.surface.dev_id;
+        renderers[id] = Some(
+            Renderer::new(&render_cx.devices[id].device, render_state.surface.format)
+                .expect("Could create renderer"),
+        );
+        Some(render_state)
+    };
+    #[cfg(not(target_arch = "wasm32"))]
+    let mut cached_window = None;
+
+    let mut scene = Scene::new();
+    let mut fragment = SceneFragment::new();
+    let mut simple_text = SimpleText::new();
+    let start = Instant::now();
+
+    let mut touch_state = multi_touch::TouchState::new();
+    let mut navigation_fingers = HashSet::new();
+    let mut transform = Affine::IDENTITY;
+    let mut mouse_down = false;
+    let mut prior_position: Option<Vec2> = None;
+    // We allow looping left and right through the scenes, so use a signed index
+    let mut scene_ix: i32 = 0;
+    if let Some(set_scene) = args.scene {
+        scene_ix = set_scene;
+    }
+    let mut prev_scene_ix = scene_ix - 1;
+    #[allow(unused)]
+    let proxy = event_loop.create_proxy();
+    // _event_loop is used on non-wasm platforms
+    event_loop.run(move |event, _event_loop, control_flow| match event {
+        Event::WindowEvent {
+            ref event,
+            window_id,
+        } => {
+            let Some(render_state) = &mut render_state else { return };
+            if render_state.window.id() != window_id {
+                return;
+            }
+            match event {
+                WindowEvent::CloseRequested => *control_flow = ControlFlow::Exit,
+                WindowEvent::KeyboardInput { input, .. } => {
+                    if input.state == ElementState::Pressed {
+                        match input.virtual_keycode {
+                            Some(VirtualKeyCode::Left) => scene_ix = scene_ix.saturating_sub(1),
+                            Some(VirtualKeyCode::Right) => scene_ix = scene_ix.saturating_add(1),
+                            Some(VirtualKeyCode::Escape) => {
+                                *control_flow = ControlFlow::Exit;
+                            }
+                            _ => {}
+                        }
+                    }
+                }
+                WindowEvent::Touch(touch) => {
+                    match touch.phase {
+                        TouchPhase::Started => {
+                            if touch.location.y > render_state.surface.config.height as f64 - 400. {
+                                navigation_fingers.insert(touch.id);
```

Comment:
Please document here that `navigation_fingers` prevents touches to be considered for gesture handling and gestures are intentionally disabled at the bottom margin of the screen as that's reserved for scene navigation.